hnn_core/parallel_backends.py

```diff
@@ -634,6 +640,10 @@ def simulate(self, net, n_trials, postproc=True):
         dpl: list of Dipole
             The Dipole results from each simulation trial
         """
+        with self.killed_lock:
+            if self.killed:
+                print("Preventing new MPIBackend simulation from starting")
+
```

Comment:
I don't quite get what's happening here. Maybe documenting the above two parameters will clarify